Add python 3.11 support

milljm opened the pull request:
Add Python 3.11 to our array of python versions to support.

Closes #26560

Commit:
Add Python 3.11 to the mix

Add Python 3.11 to our array of python versions to support.

Closes #26560

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2002325/) on 6f0ab2f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Use local moose repo

Set path source to local content.
Remove Python constraints. They are no longer necessary.

Commit:
Missing make...

huh, missing `make`. Guess we're getting lucky with all the other
recipes?

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2003616/) on 62d5515 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2003616/) on 62d5515 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
UGH. Bump peacock to allow python 3.11

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2004273/) on 5433882 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2004273/) on 5433882 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2004269/) on 5433882 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26566/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2004313/) on 5433882 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26566/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Apptainer build GCC min](https://civet.inl.gov/job/2004277/) on 5433882 : invalidated by @milljm

Error during download/extract/detection of FBLASLAPACK

milljm commented:
On Linux, there does not appear to be a solution for Python 3.11 and HDF5=1.12.x=mpi_mpich* and VTK

```pre
conda create -n testing python=3.11 vtk hdf5=1.12.1=mpi_mpich_*
```

If we bump HDF5 to 1.14, there is a solution. But historically doing so takes a lot of work.

Commit:
A linux combination

Pushing so that I can pull on a Macintosh machine to begin figuring
out Macintosh dependencies.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2005671/) on fa88b60 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2005671/) on fa88b60 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2005671/) on fa88b60 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
pushing so I can pull on another machine

Because of Zscaler, the recipe does download VTK. Need to push so
I can pull onto another machine... so lame.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006914/) on f6fe66f wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006914/) on f6fe66f wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006914/) on f6fe66f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
remove [osx] only restraint on libclang

Having a pin for libclang only for OSX targets was a mistake.

Remove petsc `make check` for now...

Add CURL_CA_BUNDLE
source:
  url: paths use `curl`, and therfore we need to pass CA certs.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2008066/) on 2a500f5 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2008066/) on 2a500f5 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented on conda/mpich/meta.yaml:
```diff
@@ -36,29 +36,26 @@ requirements:
     - {{ moose_libgfortran }}
     - {{ moose_libgfortran5 }}
-    - {{ moose_hdf5 }}
+    - hdf5=1.14.3=mpi*
```
This is now necessary. conda_build_config.yaml does not support variant wildcards (as far as I know).

Commit:
Mac Intel pinnings

And here are the Mac Intel dependency pins...

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2008177/) on 93e9e4f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad commented:
I'm curious whether you run into this failure. It has always failed for me with my system 3.11 python
```
misc/signal_handler.test_signal: Working Directory: /home/lindad/projects/PR_user_training/test/tests/misc/signal_handler
misc/signal_handler.test_signal: Running command: /home/lindad/projects/PR_user_training/test/moose_test-opt -i simple_transient_diffusion_scaled.i --error --error-override --no-gdb-backtrace
misc/signal_handler.test_signal: Python exception encountered:
misc/signal_handler.test_signal: 
misc/signal_handler.test_signal: Traceback (most recent call last):
misc/signal_handler.test_signal:   File "/home/lindad/projects/PR_user_training/python/TestHarness/schedulers/RunParallel.py", line 54, in run
misc/signal_handler.test_signal:     job.run()
misc/signal_handler.test_signal:   File "/home/lindad/projects/moose/python/TestHarness/schedulers/Job.py", line 231, in run
misc/signal_handler.test_signal:     self.__tester.run(self.timer, self.options)
misc/signal_handler.test_signal:   File "/home/lindad/projects/moose/python/TestHarness/testers/Tester.py", line 420, in run
misc/signal_handler.test_signal:     self.runCommand(timer, options)
misc/signal_handler.test_signal:   File "/home/lindad/projects/moose/python/TestHarness/testers/SignalTester.py", line 70, in runCommand
misc/signal_handler.test_signal:     self.send_signal(self.process.pid)
misc/signal_handler.test_signal:   File "/home/lindad/projects/moose/python/TestHarness/testers/SignalTester.py", line 43, in send_signal
misc/signal_handler.test_signal:     out_dupe.seek(0)
misc/signal_handler.test_signal:   File "/usr/lib/python3.11/tempfile.py", line 947, in seek
misc/signal_handler.test_signal:     return self._file.seek(*args)
misc/signal_handler.test_signal:            ^^^^^^^^^^^^^^^^^^^^^^
misc/signal_handler.test_signal: ValueError: seek of closed file
misc/signal_handler.test_signal: 
```

milljm commented:
> I'm curious whether you run into this failure. It has always failed for me with my system 3.11 python

Yes, this PR certainly does run into this... https://civet.inl.gov/job/2004305/
Just one of many issues I will have to tackle!

lindsayad commented:
I'm glad it's not just my environment 😅

Commit:
adding zlib conda package

During configure of libMesh package, we need zlib

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2009295/) on 7d43667 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2009295/) on 7d43667 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2011421/) on 82e4ccf wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2011421/) on 82e4ccf wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Only work with opened files

Add logic which allows method to continue if the file(s) are not
closed.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2014097/) on 0cb84c3 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Python 3.11 requires globals at begining

Python 3.11 now requires you to place you global regex's at the
begining of the regular expression.

milljm commented on python/TestHarness/testers/SignalTester.py:
```diff
@@ -39,15 +39,16 @@ def send_signal(self,pid):
-            #go to the beginning of the file and see if its actually started running the binary
-            out_dupe.seek(0)
-            output = out_dupe.read()
+            if not out_dupe.closed:
```
I am very concerned that this works. I would expect the inability to read the file should cause this test to fail. But it does not. So I don't understand what is going on here.

lindsayad replied:
@pbehne

pbehne replied:
Full disclosure: I worked on patching a bug with signal checkpoints, but didn't touch this send_signal function and don't think I fully understand it. That being said, shouldn't the time.sleep, os.kill, and break lines below be indented to be under the `if not out_dupe.closed` check? My assumption is that if out_dupe is already closed, then the moose binary has finished running. In that case, os.kill to a nonexistent PID should raise a ProcessLookupError. If this isn't happening, and the tests are always passing, my guess is that this code is always executed when the binary is running and out_dupe is not closed. I will poke around at this a bit more.

pbehne replied:
[pid_example.txt](https://github.com/idaholab/moose/files/14030841/pid_example.txt)

I have simplified the logic into the attached script. The 'test' is running `echo test ; sleep 1`. When the script is run with line 37 commented and line 38 uncommented (i.e., `time.sleep(5)`), os.kill is still able to send a signal to the pid even though the process should have already been terminated (the timer gives a wait time > 5 s). However, when the script is run with line 37 uncommented and line 38 commented (i.e., `os.waitpid`), os.kill is not able to send the signal because the process has already terminated. HOWEVER, the time gives a wait time on the order of 1 s, so it seems reasonable that using `time.sleep(5)` without `waitpid` would give the subprocess PLENTY of time to terminate. I am not sure how to explain this behavior. Are you? @permcody?

lindsayad replied:
I myself don't know how to explain anything. All I know is that ever since I went to python 3.11 on my machine this test has failed every time I run `run_tests` and I'll be really happy when it doesn't fail anymore

permcody replied:
This may have to do with reaping an exited process. When a process terminates it doesn’t actually end. It hangs around until the parent process closes it out. Until that occurs. The “dead” process is left in “zombie” state. We can take a look at this tomorrow and get to the bottom of it.

milljm replied:
Well the test just failed on Intel Mac (https://civet.inl.gov/job/2016334/). It usually passes.

```pre
misc/signal_handler.test_signal_parallel: Working Directory: /opt/civet1/build/moose/test/tests/misc/signal_handler
misc/signal_handler.test_signal_parallel: Running command: mpiexec -n 3 /opt/civet1/build/moose/test/moose_test-opt -i simple_transient_diffusion_scaled.i Executioner/num_steps=70 --error --error-override --timing Outputs/perf_graph=true --no-gdb-backtrace
misc/signal_handler.test_signal_parallel: 
misc/signal_handler.test_signal_parallel: 
misc/signal_handler.test_signal_parallel: Exit Code: 30
misc/signal_handler.test_signal_parallel: ################################################################################
misc/signal_handler.test_signal_parallel: Tester failed, reason: CRASH
misc/signal_handler.test_signal_parallel: 
[1.090s]     FAIL misc/signal_handler.test_signal_parallel FAILED (CRASH) [min_cpus=3]
```

permcody commented:
Let’s dive into this tomorrow

On Tue, Jan 23, 2024 at 4:40 PM Patrick Behne ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In python/TestHarness/testers/SignalTester.py
> <https://github.com/idaholab/moose/pull/26566#discussion_r1464122325>:
>
> > @@ -39,15 +39,16 @@ def send_signal(self,pid):
>
>              #first, make a true duplicate of the stdout file so we don't mess with the seek on the actual file
>              out_dupe = copy.copy(self.outfile)
> -            #go to the beginning of the file and see if its actually started running the binary
> -            out_dupe.seek(0)
> -            output = out_dupe.read()
> +            if not out_dupe.closed:
>
> pid_example.txt
> <https://github.com/idaholab/moose/files/14030841/pid_example.txt>
>
> I have simplified the logic into the attached script. The 'test' is
> running echo test ; sleep 1. When the script is run with line 37
> commented and line 38 uncommented (i.e., time.sleep(5)), os.kill is still
> able to send a signal to the pid even though the process should have
> already been terminated (the timer gives a wait time > 5 s). However, when
> the script is run with line 37 uncommented and line 38 commented (i.e.,
> os.waitpid), os.kill is not able to send the signal because the process
> has already terminated. HOWEVER, the time gives a wait time on the order of
> 1 s, so it seems reasonable that using time.sleep(5) without waitpid
> would give the subprocess PLENTY of time to terminate. I am not sure how to
> explain this behavior. Are you? @permcody <https://github.com/permcody>?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/26566#discussion_r1464122325>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXFOICI7YGVOGADF2GOU3LYQBDATAVCNFSM6AAAAABB42EGVOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQNBQGEZTAMBRG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

milljm commented:
Well the test just failed on Intel Mac (https://civet.inl.gov/job/2016334/). It usually passes.

```pre
misc/signal_handler.test_signal_parallel: Working Directory: /opt/civet1/build/moose/test/tests/misc/signal_handler
misc/signal_handler.test_signal_parallel: Running command: mpiexec -n 3 /opt/civet1/build/moose/test/moose_test-opt -i simple_transient_diffusion_scaled.i Executioner/num_steps=70 --error --error-override --timing Outputs/perf_graph=true --no-gdb-backtrace
misc/signal_handler.test_signal_parallel: 
misc/signal_handler.test_signal_parallel: 
misc/signal_handler.test_signal_parallel: Exit Code: 30
misc/signal_handler.test_signal_parallel: ################################################################################
misc/signal_handler.test_signal_parallel: Tester failed, reason: CRASH
misc/signal_handler.test_signal_parallel: 
[1.090s]     FAIL misc/signal_handler.test_signal_parallel FAILED (CRASH) [min_cpus=3]
```

Commit:
Solve vtkMPICommunicator.cxx:67  MPI had an error

As Casey joked:

"Don't use MPI, ESPECIALLY don't use MPI.......BUT maybe here, sure, use MPI."

I love it.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2021750/) on 4e5ed75 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2021750/) on 4e5ed75 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

pbehne committed:
Cleaned up logic in SignalTester.

The main part of this commit was reworking the logic in SignalTester.send_signal.
Namely, the check to see if the moose binary has started running and writing to the
output file was modified from an approach that copied and read the output file to
a much cheaper approach that simply queried the current position in the output file.

Additionally, the 'signal' parameter previously did not do anything. This has been patched.

Finally, the 'sleep_time' parameter has been removed because it was not deemed necessary to
the testing process (there should be no such thing as a signal being sent too early after the
binary has started to run; the checkpoint will simply be output at the end of the next time step).

Commit:
update to use Casey's PETSc changes

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2024143/) on aa09f7e wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2024143/) on aa09f7e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2024143/) on aa09f7e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
forgot to bump libMesh

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 7c03afd53c9c7fd7fcb6e3e20bae37cf1b124986
+Subproject commit 31948b018e9bea83c138035e952d48065458ba4a
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2031451/) on d13cc46 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2031500/) on b7a32e3 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2031500/) on b7a32e3 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
That was fun. I rebased against my origin, forgetting that I had included a merge from Casey's origin in the mix (his PETSc update), also forgetting that Casey's branch was merged into Next.

So... redid it all, rebasing against upstream/next properly.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2031549/) on 2d9b88f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2031695/) on 307c96c wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2031695/) on 307c96c wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented on scripts/tests/versioner_hashes.yaml:
```diff
@@ -189,9 +189,15 @@ d5ed287fadcea6167a6c9d75924f1c11007ab51d: #26383
-e6ada94b4f7b638bab0da284abc02315fcf63f3e: #26462
+668ed3ad53d35c108044f8f742507e07a278d48f: #26462
   mpich: 702900f
   petsc: b32e40b
-  libmesh: 7cd53c3
-  moose-dev: 4b9bf44
+  libmesh: 376999c
+  moose-dev: 82b4336
+  wasp: 91987a3
```
I don't follow... I am changing everything (mpich 4.0.2 -> 4.1.2), so everything will change (except wasp).

But I rebased against Next, and this is what I got. What am I supposed to rebase against? Do I have to cherry-pick around your merged branch or something?

cticenhour replied:
This is the entry from #26462, not this PR (which you added in Line 198 and after). Your PR shouldn't change this entry. Since my changes are now in next, you should remove commits [939a8be](https://github.com/idaholab/moose/pull/26566/commits/939a8beb511feb69b8196f87ebacbb1c1b561ddb) and [2e82820](https://github.com/idaholab/moose/pull/26566/commits/2e828204d1f82e1567aecb3de8f50f70983164b8), and then re-confirm that your build changes in [cfdb064](https://github.com/idaholab/moose/pull/26566/commits/cfdb064b98265784f7070352f9bc2d9a33fb7e11) are as you expect. The fact that the first two commits I cite above worked cleanly suggests that your rebase was on a slightly older next.

cticenhour replied:
This has been resolved through an offline conversation and interactive rebase shenanigans.

Commit:
bump versioner_hashes

cticenhour commented on framework/doc/packages_config.yml:
```diff
@@ -5,9 +5,9 @@ minimum_clang: 10.0.1
-petsc: 3.20.3
-moose_dev: 2024.01.23
-moose_tools: 2023.12.20
+petsc: 3.20.1
+moose_dev: 2024.01.15
+moose_tools: 2024.01.15
```
```suggestion
petsc: 3.20.3
moose_dev: 2024.01.24
moose_tools: 2024.01.24
```

cticenhour replied:
Resolved in fb9c0e4

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2032016/) on 8032089 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Update packages_config.yml

Update documentation template configration with new versions.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2032077/) on fb9c0e4 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2032077/) on fb9c0e4 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
Another git adventure brought to you by milljm.

loganharbour commented:
> Another git adventure brought to you by milljm.
> 
> Thank you @cticenhour for helping me sort it out!

![lizard-hehe](https://github.com/idaholab/moose/assets/22062133/c947f663-02fd-4777-b01e-0de00c0a685f)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2037737/) on c92c0dc wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2037737/) on c92c0dc wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Not sure why, but I needed to bump this

After rebasing on Next to get the latest stuff to fix Next, I find
that I need to bump rev-parse head, as my previous commit is failing
versioner unittests.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2038698/) on 1aa97f5 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
This is almost ready for 'Next' scrutiny.... Just need some help with the exodiff. @lindsayad if you would like to take a stab at it?

lindsayad commented:
wow didn't expect there to be so many!

lindsayad commented on conda/libmesh-vtk/build.sh:
```diff
@@ -18,11 +18,13 @@ cmake .. -G "Ninja" \
-    -DVTK_GROUP_ENABLE_Web:STRING=NO 
+    -DVTK_GROUP_ENABLE_Web:STRING=NO \
+    -DVTK_GROUP_ENABLE_MPI=NO \
+    -DVTK_USE_MPI:BOOL=OFF \
```
why this change?

milljm replied:
Latest MPICH mucked with this somehow. And its how these errors were cropping up all over the place:

```pre
kernels/hfem.dirichlet: 2024-01-22 06:51:22.496 (   0.059s) [         26216F0] vtkMPICommunicator.cxx:67    WARN| MPI had an error
kernels/hfem.dirichlet: ------------------------------------------------
kernels/hfem.dirichlet: Invalid datatype, error stack:
kernels/hfem.dirichlet: internal_Bcast(107): MPI_Bcast(buffer=0x16b46115c, count=1, MPI_DATATYPE_NULL, 0, comm=0x84000004) failed
kernels/hfem.dirichlet: internal_Bcast(67).: Datatype for argument datatype is a null datatype
kernels/hfem.dirichlet: ------------------------------------------------
kernels/hfem.dirichlet: Abort(526462467) on node 0 (rank 0 in comm 432): application called MPI_Abort(comm=0x84000004, 526462467) - process 0
```

Doing some research on this error turned up rather dull one-offs. But it Kitware blames this error on users not using the same version of MPI as it was built with. Of course, that was not our case. But it was the only reference I could find on the internets.

milljm replied:
The line right below your comment:

`-DVTK_MODULE_ENABLE_VTK_IOMPIParallel=WANT`

We're still using MPI. But only for the necessary FileWriter we are using in libMesh (or so John Peterson required back in the day).

lindsayad replied:
sounds good

lindsayad commented on conda/mpich/meta.yaml:
```diff
@@ -53,9 +53,9 @@ requirements:
+    - {{ moose_libclang13 }}
     - {{ moose_ld64 }}          # [osx]
-    - {{ moose_libclang }}      # [osx]
-    - {{ moose_libclang13 }}    # [osx]
```
how does this libclang versioning work? Previously we were using clang 12 but we link libclang13, and now clang 15 is still linked with libclang13?

lindsayad commented:
These failures are crazy. It's like MOOSE meshing has changed with this PR but only on ARM even though this doesn't touch PETSc/libMesh/MOOSE code. Tests are even failing in serial so it's not the mpich bump either

roystgnr commented:
I just looked at the gold/partial_circle_rad_in.e from the meshgenerators/circular_correction_generator.partial_curve_rad failure, and it's definitely not a robust test.  The mesh is being generated via XYDelaunay, and the trouble with "generate a Delaunay triangulation of this boundary" in regression tests is that typically the result is *not* unique.

Looking at the bottom right corner of the mesh, where the exodiff complains, we can see nodes at (0,-0.8), (0,-0.9), (0,-1), (-0.1,-1), (-0.2,-1).  The three bottom right nodes are going to form one triangle ... and then the remaining part of those nodes' convex hull is a perfectly symmetrical trapezoid.  In the gold solution, the trapezoid is divided by the diagonal connecting (-0.2,-1) to (0,-0.9), and that's a Delaunay mesh, but based on the exodiff the failing solution divides that trapezoid by the diagonal connecting (-0.1,-1) to (0,-0.8), and technically that's still a success because that's *also* a Delaunay mesh.  Yay symmetry.

I'm not sure what the best solution is here.  When I created my own XYDelaunay tests I tried to add asymmetries to the domain to prevent any ambiguity from showing up in them ... but Civet is saying that two of my own tests have failed too!?

roystgnr commented:
What the hell.  I just looked at the diffs on this PR.  I stand by my explanation of why those tests might get knocked over by a feather, but **where is the feather**?  There should at least be *something* changing the executables, the underlying malloc/free, *something* that could get us iterating over elements in a different order.

lindsayad commented:
example stack traces from the parallel hangs:

<details>
<summary>process 0</summary>
<pre>
    frame #14: 0x000000010196cfd4 libmpifort.12.dylib`mpi_allreduce_ + 144
    frame #15: 0x000000010af695b8 libpetsc.3.20.dylib`mumps_propinfo_ + 72
    frame #16: 0x000000010ae96b5c libpetsc.3.20.dylib`dmumps_solve_driver_ + 15164
    frame #17: 0x000000010aef86f4 libpetsc.3.20.dylib`dmumps_ + 1748
    frame #18: 0x000000010aefda2c libpetsc.3.20.dylib`dmumps_f77_ + 6892
    frame #19: 0x000000010aef683c libpetsc.3.20.dylib`dmumps_c + 3552
    frame #20: 0x000000010a5e3ff4 libpetsc.3.20.dylib`MatSolve_MUMPS + 688
    frame #21: 0x000000010a60f3c4 libpetsc.3.20.dylib`MatSolve + 292
    frame #22: 0x000000010a9b5f6c libpetsc.3.20.dylib`PCApply_LU + 84
    frame #23: 0x000000010a99f96c libpetsc.3.20.dylib`PCApply + 204
    frame #24: 0x000000010a9a15ec libpetsc.3.20.dylib`PCApplyBAorAB + 632
    frame #25: 0x000000010aba0af0 libpetsc.3.20.dylib`KSPSolve_GMRES + 1112
    frame #26: 0x000000010ab48aac libpetsc.3.20.dylib`KSPSolve_Private + 1056
    frame #27: 0x000000010ab48648 libpetsc.3.20.dylib`KSPSolve + 16
    frame #28: 0x000000010abda684 libpetsc.3.20.dylib`SNESSolve_NEWTONLS + 1316
    frame #29: 0x000000010ac18ff8 libpetsc.3.20.dylib`SNESSolve + 1372
</pre>
</details>

<details>
<summary>process 1</summary>
<pre>
    frame #14: 0x0000000104cb0fd4 libmpifort.12.dylib`mpi_allreduce_ + 144
    frame #15: 0x000000010e2b98b0 libpetsc.3.20.dylib`mumps_sol_rhsmapinfo_ + 160
    frame #16: 0x000000010e1debdc libpetsc.3.20.dylib`dmumps_solve_driver_ + 31676
    frame #17: 0x000000010e23c6f4 libpetsc.3.20.dylib`dmumps_ + 1748
    frame #18: 0x000000010e241a2c libpetsc.3.20.dylib`dmumps_f77_ + 6892
    frame #19: 0x000000010e23a83c libpetsc.3.20.dylib`dmumps_c + 3552
    frame #20: 0x000000010d927ff4 libpetsc.3.20.dylib`MatSolve_MUMPS + 688
    frame #21: 0x000000010d9533c4 libpetsc.3.20.dylib`MatSolve + 292
    frame #22: 0x000000010dcf9f6c libpetsc.3.20.dylib`PCApply_LU + 84
    frame #23: 0x000000010dce396c libpetsc.3.20.dylib`PCApply + 204
    frame #24: 0x000000010dce55ec libpetsc.3.20.dylib`PCApplyBAorAB + 632
    frame #25: 0x000000010dee4af0 libpetsc.3.20.dylib`KSPSolve_GMRES + 1112
    frame #26: 0x000000010de8caac libpetsc.3.20.dylib`KSPSolve_Private + 1056
    frame #27: 0x000000010de8c648 libpetsc.3.20.dylib`KSPSolve + 16
    frame #28: 0x000000010df1e684 libpetsc.3.20.dylib`SNESSolve_NEWTONLS + 1316
    frame #29: 0x000000010df5cff8 libpetsc.3.20.dylib`SNESSolve + 1372
</pre>
</details>

milljm commented:
I don't know much about troubleshooting the hangs, but what I can tell you is that non-INL profile'd machines  (no JAMF), seem to not hang. -my personal Apple Si machine is fine. But I still get the same exodiff that Civet is reporting.

lindsayad commented:
> I don't know much about troubleshooting the hangs, but what I can tell you is that non-INL profile'd machines (no JAMF), seem to not hang. -my personal Apple Si machine is fine.

It sounds like most recently on your personal machine you do get the hang though right? That was without your newest mpich stack?

milljm commented:
> > I don't know much about troubleshooting the hangs, but what I can tell you is that non-INL profile'd machines (no JAMF), seem to not hang. -my personal Apple Si machine is fine.
> 
> It sounds like most recently on your personal machine you do get the hang though right? That was with your newest mpich stack?

Yeah, crazy! This is the first I've seen it hang on non-INL equipment. Simple hello-world examples run to completion just fine.

milljm commented:
Closing in favor of #26839, which includes (will include) all the Python 3.11 fixes
